crates/autopilot/src/domain/competition/mod.rs

```diff
@@ -41,7 +41,7 @@ impl SolutionWithId {
         self.solution.score()
     }
 
-    pub fn order_ids(&self) -> impl Iterator<Item = &domain::OrderUid> {
+    pub fn order_ids(&self) -> impl Iterator<Item = &domain::OrderUid> + std::fmt::Debug {
```

Comment:
By doing this we are enforcing that the `Iterator` type implements `Debug`. If the need to change the underlying iterator to add a `map` or `filter`, this code will fail. Another solution would be adding `.collect::<Vec<_>>()` to the debug line, but it allocates...

Reply:
The iterator returned by `.filter()` and `.map()` still implements `Debug` so this would not cause issues.